### PR TITLE
feat(database): sets up db, adds movie model, and model unit tests

### PR DESCRIPTION
**Pull request by jaonjohnkim**
**What:** 
- Adds bookshelf, pg, and knex
- Includes script for seeding, testing, and migrating.

**Why:**
This step allows connecting to the database and adds aliases to seed, migrate, and test the database.

**Context:**
The previous PR sets up the basic hapi server but no endpoints.
Now with the db set up, the next PR will have endpoint configuration

### CODE REVIEW

File: db/knexfile.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// 'use strict';
```
Comment (Lianathanoj) [inline db/knexfile.js]: Do you think we need this file? I believe `db/index.js` already acts as the knexfile. When submitting a PR, you want your file changes to not have any extraneous aspects, e.g. an unneeded file that is commented out. As you can see from our codebase, we also don't use comments often (although Kienan was testing out a new documentation procedure for our enqueue endpoint [here](https://github.com/lob/lob-api/commit/b09674c4b3471fc19d0a56f55ea98ce7d76fc024)) unless we want to document something more specific. I believe our current philosophy revolves around self-documenting code for the most part.

**Comment by Lianathanoj** [comment]
As a rule of thumb, we want to split up our PRs to ensure modularity and to ensure we aren't reviewing a behemoth of information. While this isn't a problem right now, I would recommend in the future splitting up your PRs into smaller steps if you realize there are unrelated steps in the same PR. 

In this case, I would have separated 1) setting up the db and 2) adding the movie model and model unit tests. I think it's fine for now, but make sure you split up your PRs accordingly. Don't be like me and make a 500 line PR for adding like 4 endpoints!

**Comment by Lianathanoj** [comment]
Another comment: you want to try writing good PR descriptions at this point. I know this is Miyagi and there isn't much to say about these earlier steps, but we ascribe to a convention where we might have sections such as **What, Why, Details, Context, Notes,** etc. (although sometimes not all sections are necessary) to ensure that reviewers get a good amount of context. Take a look at any of the PRs in our API repo; you'll always see some good examples of PR descriptions.

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,10 @@ node_js:
   - 6.11.1
 env:
   - NODE_ENV="test"
+before_script:
+  - yarn run db:setup:user
```
Comment (Lianathanoj) [inline .travis.yml]: Nit: I think you don't need `run` here or in the next line. I believe you would have needed `npm run` for npm, but for yarn, you can simply have it as `yarn db:setup:user`.

File: db/knexfile.js (hunk 1 above)
Comment (jaonjohnkim) [reply db/knexfile.js]: Oh yea, when I was reading through the Miyagi, I first created this and then retrospectively realized that the index.js was the replacement. Forgot to delete. Thanks for the catch!

File: .travis.yml (hunk 2 above)
Comment (jaonjohnkim) [reply .travis.yml]: Yea, I thought so too. Miyagi includes `run` for some reason. Will change it

File: db/migrations/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "lob/migrations"
+}
```
Comment (sjl2) [inline db/migrations/.eslintrc]: nit: can you add a newline here?

File: package.json [hunk 4]
```diff
@@ -20,6 +24,13 @@
   },
   "scripts": {
     "enforce": "istanbul check-coverage --statement 100 --branch 100 --function 100 --lines 100",
```
Comment (sjl2) [inline package.json]: nit: can you alphabetize the script fields? There should be a tool to do it in your editor.

Comment (sjl2) [reply package.json]: That said, it looks like it's just `enforce` that needs to be moved down.

**Comment by sjl2** [comment]
Another thing is you want to split up the DB migrations with the application code that depends on those changes. When we deploy our code, the application code takes effect before the migration is ran. 

That means if a PR has code that depends on a migration in the same PR, it will cause the application to error until the migration is run. For some services (and SF Movies) this is fine, particularly if it's not being used between the deployment and the migration. However, some services like API will 500 for customers if this is the case.

You'll learn more about API deployment during your first API tasks, but in the mean time, you'll want to think about splitting up your PRs into the bare minimum that introduces new functionality.

File: db/migrations/.eslintrc (hunk 3 above)
Comment (jaonjohnkim) [reply db/migrations/.eslintrc]: done!

**Pull request merged**